packages/runtime/src/widgets/Form/MultiSelect.tsx

```diff
@@ -84,7 +84,12 @@ const MultiSelect: React.FC<MultiSelectProps> = (props) => {
       isArray(values?.initialValue)
     ) {
       prevInitialValue.current = values?.initialValue || [];
-      setSelectedValues(values?.initialValue);
+      const initialValue = values?.initialValue.map((item) =>
+        isObject(item)
+          ? (get(item, values.valueKey || "value") as string)
+          : item,
```

Comment:
I'm not sure why you are coercing as a string? The value itself should be able to be an object.